tests/bluetooth/tester/src/btp_mesh.c

```diff
@@ -162,13 +162,118 @@ static int dfd_srv_send(struct bt_mesh_dfd_srv *srv,
 	return 0;
 }
 
+#ifdef CONFIG_BT_MESH_DFD_SRV_OOB_UPLOAD
+static struct {
+	uint8_t uri[CONFIG_BT_MESH_DFU_URI_MAXLEN];
+	uint8_t uri_len;
+	uint8_t fwid[CONFIG_BT_MESH_DFU_FWID_MAXLEN];
+	uint8_t fwid_len;
+	const struct bt_mesh_dfu_slot *slot;
+	uint8_t progress;
+	bool started;
+} dfd_srv_oob_ctx;
+
+static void oob_check_handler(struct k_work *work);
+static K_WORK_DEFINE(oob_check, oob_check_handler);
+static void oob_store_handler(struct k_work *work);
+static K_WORK_DEFINE(oob_store, oob_store_handler);
+
+static int dfd_srv_start_oob_upload(struct bt_mesh_dfd_srv *srv,
+				    const struct bt_mesh_dfu_slot *slot,
+				    const char *uri, uint8_t uri_len,
+				    const uint8_t *fwid, uint16_t fwid_len)
+{
+	LOG_DBG("Start OOB Upload");
+
+	memcpy(dfd_srv_oob_ctx.uri, uri, uri_len);
+	dfd_srv_oob_ctx.uri_len = uri_len;
+	memcpy(dfd_srv_oob_ctx.fwid, fwid, fwid_len);
+	dfd_srv_oob_ctx.fwid_len = fwid_len;
+	dfd_srv_oob_ctx.slot = slot;
+	dfd_srv_oob_ctx.progress = 0;
+	dfd_srv_oob_ctx.started = true;
+
+	k_work_submit(&oob_check);
+
+	return BT_MESH_DFD_SUCCESS;
+}
+
+static void dfd_srv_cancel_oob_upload(struct bt_mesh_dfd_srv *srv,
+				      const struct bt_mesh_dfu_slot *slot)
+{
+	LOG_DBG("Cancel OOB Upload");
+
+	dfd_srv_oob_ctx.started = false;
+}
+
+static uint8_t dfd_srv_oob_progress_get(struct bt_mesh_dfd_srv *srv,
+					const struct bt_mesh_dfu_slot *slot)
+{
+	uint8_t progress;
+
+	if (dfd_srv_oob_ctx.started) {
+		progress = dfd_srv_oob_ctx.progress;
+
+		dfd_srv_oob_ctx.progress = MIN(dfd_srv_oob_ctx.progress + 25, 99);
+
+		if (dfd_srv_oob_ctx.progress == 99) {
```

Comment:
Should this be >= or <=, or is it correct with exact 99. dfd_srv_oob_ctx.progress can be less than 99.

Reply:
ASFAIK The image is ready for storing if progress is exactly 99. After storing it will be 100.